internal/form/repo.go

```diff
@@ -72,6 +72,9 @@ func (f MigrateRepo) ParseRemoteAddr(user *db.User) (string, error) {
 		if len(f.AuthUsername)+len(f.AuthPassword) > 0 {
 			u.User = url.UserPassword(f.AuthUsername, f.AuthPassword)
 		}
+		if u.Scheme == "git" && u.Port() != "" && (strings.Contains(remoteAddr, "%0d") || strings.Contains(remoteAddr, "%0a")) {
```

Comment:
Why is checking `u.Scheme` and `u.Port()` necessary? Shouldn't we disallow malicious chars in all cases?

Reply:
I wouldn't say it is necessary, however other protocols have this filtered properly so I guess it is ok to proceed as is. Do you have any alternative plan? By default, git:// protocol will only use 9418 port for transmitting the data.

Reply:
I see... checking the protocol makes sense now. As for port, can I use urlencoded new lines without specifying a port?

Reply:
yes. there seems no problem assuming that nobody uses crlf for a repository name.

Reply:
What I meant was we always need to check _urlencoded new lines_, with or without having a port in the clone URL, the port doesn't play a role here. Does this sound correct to you?

Reply:
Yes. That sounds correct. We're on the same page.
If you think there's no need of applying the port number, I think it's fine to delete it in this case.

Initially, this will only cause some impact when the attacker is capable of attacking other ports. 
But as you mentioned, blocking all ports with %0d and %0a also sounds good to me.

Reply:
Updated: https://github.com/gogs/gogs/pull/6420/commits/eea3d678561b885cbcce3c87077927d308abe340